### PR TITLE
add basic howto docs for publishing

### PR DESCRIPTION
**Pull request by rhelmer**
@hamilton as promised, these are the docs outlining the steps I am taking to publish the add-ons. These have a lot of TODOs and might have some minor mistakes :) But I thought it was important to have a good starting point.

### CODE REVIEW

File: docs/IonCorePublishing.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Ion Core Extension Publishing HOWTO
+
+## First-time
+
+Since the core extension requires special Firefox-only privileges, it must use the Mozilla signing infrastructure.
+This requires that the github repo is cloned to the `github.com/mozilla-extensions` organization, and have a working `.taskcluster.yml`
```
Comment (hamilton) [inline docs/IonCorePublishing.md:6]:
```suggestion
This requires that the github repo is cloned to the `github.com/mozilla-extensions` organization, and have a working `.taskcluster.yml` file in the repository.
```

File: docs/IonCorePublishing.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+The Mozilla Taskcluster CI will then automatically start working for this new clone.
+
+NOTE - this must be a clone and not a fork (TODO - file/link to bug on TC side about this?)
+
+## Finding the QA-signed extension
+
+A good CI run will look like:
```
Comment (hamilton) [inline docs/IonCorePublishing.md:14]: how does one get to this url? Through the github actions? Something else?

File: docs/IonCorePublishing.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+The "dep-signing-ion_core" task will link to the completed build on Taskcluster:
+https://firefox-ci-tc.services.mozilla.com/tasks/ZcZWnI5pT4K_1ycCX3l7iw
+
+Click the "Artifacts" link to find a link to the signed XPI `public/build/ion_core.xpi`.
+
+Producing a production-signed extension
+---------------------------------------
+
+Production signing requires sign-off from addons and releng.
+See the #addons-pipeline channel on Slack for assitance.
```
Comment (hamilton) [inline docs/IonCorePublishing.md:26]:
```suggestion
See the #addons-pipeline channel on Slack for assistance.
```

File: docs/StudyPublishing.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Study Extension HOWTO
+
+## Getting Started
+
+Starting with a github repository containing your extension (such as https://github.com/mozilla-ion/ion-basic-study):
```
Comment (hamilton) [inline docs/StudyPublishing.md:5]:
```suggestion
Starting with a github repository containing your study extension (such as [the Ion Basic Study](https://github.com/mozilla-ion/ion-basic-study)):
```

File: docs/StudyPublishing.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+- create account on https://addons.mozilla.org
+- run `web-ext build` to generate your extension in `web-ext-artifacts/`
+- upload your extension
```
Comment (hamilton) [inline docs/StudyPublishing.md:9]:
```suggestion
- upload the resulting `.zip` file in `web-ext-artifacts/` to addons.mozilla.org
```

File: docs/StudyPublishing.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+It may take some time for your extension to be reviewed and available for download.
+
+## Update RemoteSettings Server
+
+- The Ion Core add-on checks the Firefox RemoteSettings server for a list of the currently-approved studies.
+- For local testing, modify `public/locally-available-studies.json` and `npm run local-build-addon`, then `web-ext run --pref=extensions.experiments.enabled=true -f nightly`
+    - TODO link to README.md instead of repeating ourselves?
+- There is also a RemoteSettings dev server you may use for local testing / QA
+    - https://remote-settings.readthedocs.io/en/latest/tutorial-dev-server.html
+- Updates to the staging and production servers must recieve sign off
```
Comment (hamilton) [inline docs/StudyPublishing.md:20]:
```suggestion
- Updates to the staging and production servers must receive sign off
```

**Review by hamilton — APPROVED** [review]
Looks great – thanks for writing this up! Some small fixes & clarifications left as suggestions that you can commit from the files view

**Comment by hamilton** [comment]
@rhelmer one additional note – typically markdown files I've seen in GH take on `THIS_KIND_OF_FORMAT.md` for the title. Happy to go with whatever but just noting that this is usually the md file structure used

File: docs/IonCorePublishing.md (hunk 2 above)
Comment (rhelmer) [reply docs/IonCorePublishing.md:14]: looks like it's a github action: https://github.com/mozilla-extensions/ion-core-addon/runs/1261759243

**Comment by rhelmer** [comment]
> @rhelmer one additional note – typically markdown files I've seen in GH take on `THIS_KIND_OF_FORMAT.md` for the title. Happy to go with whatever but just noting that this is usually the md file structure used

Ah yeah let's rename, the least surprise the better.

**Comment by rhelmer** [comment]
I made some pretty significant edits @hamilton would you mind taking a second look?

**Comment by Dexterp37** [comment]
> 
> 
> @rhelmer one additional note – typically markdown files I've seen in GH take on `THIS_KIND_OF_FORMAT.md` for the title. Happy to go with whatever but just noting that this is usually the md file structure used

Oh, I wasn't aware of this. Almost all the repos I've seen at Mozilla use "kebab-case.md", except for README, CODE_OF_CONDUCT and LICENSE file (which, are uppercase for [historical reasons](http://catb.org/jargon/html/R/README-file.html)).

**Comment by rhelmer** [comment]
In the interest of having a starting point I'm going to land as-is, it's easy to rename. Maybe best to standardize a consistent structure around layout of docs and then make this conform.

**Pull request merged**